bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -787,8 +825,10 @@ void ReplicaImp::onMessage<PrePrepareMsg>(PrePrepareMsg *msg) {
     return;
   }
 
-  if (!validatePreProcessedResults(msg)) {
-    // validatePreProcessedResults() logs the error
+  if ((!getReplicaConfig().prePrepareFinalizeAsyncEnabled) && (!validatePreProcessedResults(msg))) {
+    // trigger view change
+    LOG_ERROR(VC_LOG, "PreProcessResult Signature failure. Ask to leave " << getCurrentView());
```

Comment:
-> Ask to leave view

Reply:
Thanks @yuliasherman for your comment, I have incorporated the changes.